tsl/src/compression/compression.c

```diff
@@ -1590,7 +1590,33 @@ row_decompressor_decompress_row(RowDecompressor *decompressor, Tuplesortstate *t
 							0 /*=options*/,
 							decompressor->bistate);
 
-				ts_catalog_index_insert(decompressor->indexstate, decompressed_tuple);
+				EState *estate = CreateExecutorState();
```

Comment:
If there is any possibility, I would like it if we could avoid creating the estate for every index insertion. Maybe caching it on the decompressor level and reusing is the way to go.

Reply:
I made some changes wrt this, now we cache the estate in the decompressor.